File: baekjoon/2주차/10845-큐/창섭/10845.kt
import java.util.*

fun main() = with(System.`in`.bufferedReader()) {
    val T = readLine().toInt()
    var q = IntArray(10000)
    var front =0;
    var rear = -1;
    for(i in 0 until T) {
        var command = readLine().split(" ")
        when (command[0]) {
            "push" -> q[++rear] = command[1].toInt()
            "pop" -> if (front == rear + 1) println(-1) else println(q[front++])
            "size" -> println(rear - front + 1)
            "empty" -> if (front == rear + 1) println(1) else println(0)
            "front" -> if (front == rear + 1) println(-1) else println(q[front])
            "back" -> if (front == rear + 1) println(-1) else println(q[rear])
        }
    }
}

File: baekjoon/2주차/10845-큐/우섭/10845.py
import sys

getN = int(sys.stdin.readline())
stack = []
for _ in range(getN):
    input = sys.stdin.readline()
    if 'push' in input:
        stack.append(int(input.split()[1]))
    elif 'pop' in input:
        print(stack.pop(0) if len(stack) > 0 else -1)
    elif 'size' in input:
        print(len(stack))
    elif 'empty' in input:
        print(1 if len(stack) == 0 else 0)
    elif 'front' in input:
        print(stack[0] if len(stack) != 0 else -1)
    elif 'back' in input:
        print(stack[-1] if len(stack) != 0 else -1)
